protocols/kad/src/behaviour.rs

```diff
@@ -147,6 +147,8 @@ pub struct KademliaConfig {
     connection_idle_timeout: Duration,
     kbucket_inserts: KademliaBucketInserts,
     caching: KademliaCaching,
+    // Set the peer mode.
+    client: bool,
```

Comment:
Let's use a separate `enum` here, e.g. `enum Mode { Client, Server }` for the sake of type safety. `Mode` can then implement `Default`, which would default to `Server` (for now).